gamepad/src/gamepad.java

```diff
@@ -33,18 +36,29 @@ public static void main(String[] args) throws MqttException {
                     MqttMessage msgA = new MqttMessage("forward".getBytes());
                     msgA.setQos(0);
                     msgA.setRetained(true);
-                    publisher.publish("SimonDrives", msgA);
+                    if(!meddelande.equals(msgA) ) {
+                        publisher.publish("SimonDrives", msgA);
+                        meddelande = msgA;
+                    }
                 } else if (currController.isButtonPressed(ControllerButton.X)) {
                     MqttMessage msgX = new MqttMessage("backward".getBytes());
                     msgX.setQos(0);
                     msgX.setRetained(true);
-                    publisher.publish("SimonDrives", msgX);
+                    if(!meddelande.equals(msgX) ) {
+                        publisher.publish("SimonDrives", msgX);
+                        meddelande = msgX;
+                    }
+
                 }
                 else {
                     MqttMessage msgStop = new MqttMessage("stop".getBytes());
                     msgStop.setQos(0);
                     msgStop.setRetained(true);
-                    publisher.publish("SimonDrives", msgStop);
+                    if(!meddelande.equals(msgStop) ) {
+                        publisher.publish("SimonDrives", msgStop);
+                        meddelande = msgStop;
+                    }
+
                 }
 
                 if (currController.getAxisState(ControllerAxis.LEFTX) < -0.5){
```

Comment:
Shouldn't the if(!meddelande.equals(msg)) be happening for the AxisStates as well? 

I think the Axis states should execute this if statement as well in order to limit how often we publish to the broker. We don't wanna overload the broker

Reply:
It might also be a good idea to change the variable name meddelande to something else